nio/block/mixins/persistence/tests/test_persistence.py

```diff
@@ -124,3 +124,92 @@ def test_block_loading_from_persistence(self):
         # assert that data matches what was persisted outside of block
         self.assertEqual(block._to_be_saved, 3)
         self.assertEqual(block._to_be_saved_again, 4)
+
+
+class SerializingBlock(Persistence, Block):
+
+    def __init__(self):
+        super().__init__()
+        self._attr1 = "attr1 default"
+        self._attr2 = "attr2 default"
+        # attr3 is ignored in serialize/deserialize yet considered in
+        # persisted_values, used to show that it is ignored, since
+        # serialize/deserialize takes precedence
+        self._attr3 = "attr3 default"
+        self._not_to_be_saved = 'not_to_be_saved'
+
+    def serialize(self):
+        """ Overridden to define what is persisted """
+        return {
+            "attr1": self._attr1,
+            "attr2": self._attr2
+        }
+
+    def deserialize(self, data):
+        """ Overridden to define what is loaded """
+        self._attr1 = data["attr1"]
+        self._attr2 = data["attr2"]
+
+    def persisted_values(self):
+        """ Defined just to show precedence
+
+        It will not be saved
+        """
+        return ["_attr3"]
+
+
+class TestSerialize(NIOBlockTestCase):
+
+    def test_configures_from_scratch(self):
+        """ Tests that block configures even when there is nothing to load
+
+        Shows that when block is first instantiated and configured with no data
+        previously persisted no errors occur
+        """
+        block = SerializingBlock()
+        block_id = "test_block1"
+
+        self.configure_block(block, {"id": block_id})
+        self.assertEqual(block._not_to_be_saved, "not_to_be_saved")
```

Comment:
This test makes no assertion that nothing was actually loaded from persistence though. It should at least check to make sure that the default values of the block are still there and haven't been overwritten by deserialize.